test/env.jl

```diff
@@ -63,3 +63,8 @@ end
 
 # Test for #10853
 @test withenv(Dict{Any,Any}()...) do; true; end
+
+# Test for #18141
+for (k, v) in ENV
+    @test v[end] != '\0'
```

Comment:
Could anyone have this be the case in their pre-existing environment, or would our getenv always be reading based on null-termination?